### PR TITLE
Add metadata to user search results

### PR DESCRIPTION
**Pull request by guivinicius**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
On https://github.com/thepracticaldev/dev.to/pull/7093 I suggested adding some metadata attributes to `Search::User` so we could easily paginate the results.

This is a suggestion based on `Kaminari` pagination attributes.

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/pull/7093

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/services/search/user.rb [hunk 1]
```diff
@@ -12,31 +13,24 @@ def search_documents(params:)
...
-          "path" => hit["path"],
-          "id" => hit["id"],
-          "class_name" => "User",
-          "positive_reactions_count" => hit["positive_reactions_count"],
-          "comments_count" => hit["comments_count"],
-          "badge_achievements_count" => hit["badge_achievements_count"],
-          "last_comment_at" => hit["last_comment_at"],
-          "roles" => hit["roles"]
-        }
+        SERIALIZER_CLASS.attributes_to_serialize.keys.reduce({}) do |m, k|
```
Comment (guivinicius) [inline app/services/search/user.rb:31]: I'm sure this change breaks some place. Still need to find out.

Comment (mstruve) [reply app/services/search/user.rb:31]: I don't think we can do this bc we have key combos like `"title" => hit["name"],` and `"name" => hit["username"]`  that don't match up exactly but are needed for the main feed.

File: app/services/search/classified_listing.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class ClassifiedListing < Base
     INDEX_NAME = "classified_listings_#{Rails.env}".freeze
     INDEX_ALIAS = "classified_listings_#{Rails.env}_alias".freeze
     MAPPINGS = JSON.parse(File.read("config/elasticsearch/mappings/classified_listings.json"), symbolize_names: true).freeze
-    DEFAULT_PAGE = 0
+    DEFAULT_PAGE = 1
```
Comment (mstruve) [inline app/services/search/classified_listing.rb:6]: If we change these defaults we then have to change what the frontend is passing to the controllers bc they are all passing 0 right now.

**Comment by guivinicius** [comment]
@mstruve @rhymes

I got some time today to take a look at that again but I couldn't find where this is been used. Looks like all Elasticsearch is pointing to Algolia and doesn't hit `SearchController#users` or `SearchController#feed_content`.

If you guys could give me a better direction. That would be helpful. Thanks.

**Comment by mstruve** [comment]
One spot we currently starting paging at zero is for [searching classified listings.](https://github.com/thepracticaldev/dev.to/blob/master/app/javascript/listings/listings.jsx#L73) We [removed the user search code](https://github.com/thepracticaldev/dev.to/pull/7015/files) that was using it before which is what I was originally thinking about. I also implemented it in our new [feed search code that will serve the feed through Elasticsearch](https://github.com/thepracticaldev/dev.to/pull/7231).

**Comment by guivinicius** [comment]
Now I got it ... the changes are still on course on this part. I'm going to make skin off the body changes and just leave the metadata, that way won't break anything. 

But I really think we might need to make some changes to the paging at zero.

**Comment by guivinicius** [comment]
I'm closing this PR since I beleive this change could affect more things than I expected during this Algolia migration. I can come back later on for this. :)

**Pull request closed without merging**